tests/unit/integrations/kubeflow/orchestrators/test_kubeflow_orchestrator.py

```diff
@@ -71,3 +74,25 @@ def test_kubeflow_orchestrator_stack_validation(
             artifact_store=local_artifact_store,
             container_registry=local_container_registry,
         ).validate()
+
+
+def test_skip_ui_daemon_provisioning():
+    """Tests that the UI daemon provisioning is skipped if either set explicitly
+    or when a hostname is specified."""
+    empty_config = KubeflowOrchestratorConfig()
+    assert (
+        _get_kubeflow_orchestrator(empty_config).skip_ui_daemon_provisioning
+        is False
+    )
+
+    config = KubeflowOrchestratorConfig(
+        kubeflow_hostname="https://arias-kubeflow.com/pipeline"
+    )
+    assert (
+        _get_kubeflow_orchestrator(config).skip_ui_daemon_provisioning is True
+    )
+
+    config = KubeflowOrchestratorConfig(skip_ui_daemon_provisioning=True)
+    assert (
+        _get_kubeflow_orchestrator(config).skip_ui_daemon_provisioning is True
+    )
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "kubeflow_hostname,skip_ui_daemon_provisioning",
    [
        (None, False),
        ("https://arias-kubeflow.com/pipeline", True),
        (True, True),
    ],
)
def test_skip_ui_daemon_provisioning(
    kubeflow_hostname, skip_ui_daemon_provisioning
):
    """Tests that the UI daemon provisioning is skipped if either set explicitly
    or when a hostname is specified."""
    config = KubeflowOrchestratorConfig(
        kubeflow_hostname=kubeflow_hostname,
        skip_ui_daemon_provisioning=skip_ui_daemon_provisioning,
    )
    orchestrator = _get_kubeflow_orchestrator(config=config)
    assert (
        orchestrator.skip_ui_daemon_provisioning == skip_ui_daemon_provisioning
    )
```
Do you think we can use more of `mark.parametrize` in test functions where we have different possibilities to test or is there some problem with it?

Reply:
Good idea, I'll need three values though (hostname, skip in the config and the "combined" skip of the orchestrator). I'll update it 👍